app/src/main/java/com/khinthirisoe/bakingapp/ui/baking/view/BakingActivity.kt

```diff
@@ -0,0 +1,81 @@
+package com.khinthirisoe.bakingapp.ui.baking.view
+
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import android.widget.ProgressBar
+import android.widget.Toast
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.khinthirisoe.bakingapp.R
+import com.khinthirisoe.bakingapp.data.model.Recipe
+import com.khinthirisoe.bakingapp.di.component.AppComponent
+import com.khinthirisoe.bakingapp.di.component.DaggerActivityComponent
+import com.khinthirisoe.bakingapp.di.module.ActivityModule
+import com.khinthirisoe.bakingapp.ui.baking.BakingContract
+import com.khinthirisoe.bakingapp.ui.base.BaseActivity
+import com.khinthirisoe.bakingapp.ui.ingredients.IngredientsActivity
+import javax.inject.Inject
+
+
+class BakingActivity : BaseActivity(), BakingContract.View, BakingAdapter.BakingRecyclerViewClickListener {
+
+    @Inject
+    lateinit var presenter: BakingContract.Presenter
+
+    private lateinit var bakingRecyclerView: RecyclerView
+    private lateinit var progressBar: ProgressBar
+    private var bakingAdapter: BakingAdapter? = null
+
+    override fun setupComponent(appComponent: AppComponent) {
+        val component = DaggerActivityComponent.builder()
+            .appComponent(appComponent)
+            .activityModule(ActivityModule(this))
+            .build()
+        component.inject(this)
+
+        presenter.onAttachView(this)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(com.khinthirisoe.bakingapp.R.layout.activity_baking)
+
+        initView()
+
+        presenter.fetchBakingRecipes()
+    }
+
+    private fun initView() {
+
+        bakingRecyclerView = findViewById(R.id.recyclerView)
+        progressBar = findViewById(R.id.progressBar)
+
+        val layoutManager = LinearLayoutManager(this)
+        bakingRecyclerView.layoutManager = layoutManager
+
+        bakingAdapter = BakingAdapter(this, null, this)
+
+    }
+
+    override fun showBakingLists(recipes: ArrayList<Recipe>) {
+        bakingAdapter?.setRecipes(recipes)
+        bakingRecyclerView.adapter = bakingAdapter
+    }
+
+    override fun listItemClick(baking: Recipe) {
+        startActivity(Intent(this, IngredientsActivity::class.java).putExtra(IngredientsActivity.EXTRA_BAKING, baking))
```

Comment:
@KhinThiriSoe @KhinThiriSoe add **createIntent** method in IngredientsActivity and create the intent from there.
Make this a practice. When you need to create an activity intent, you create that intent from the **createIntent** method from the activity itself.
Same thing with fragments, you can write **newInstance** method for fragments.
This will help you a lot when your app grows

Reply:
@KhinThiriSoe have you fixed this? Where's your commit?

Reply:
still waiting for the fix